File: app/src/androidTest/java/com/vgrec/espressoexamples/DateTimePickerTest.java
package com.vgrec.espressoexamples;

import android.widget.DatePicker;
import android.widget.TimePicker;

import androidx.test.espresso.contrib.PickerActions;

import com.vgrec.espressoexamples.bases.TestBase;

import org.hamcrest.Matchers;
import org.junit.Test;

import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.action.ViewActions.click;
import static androidx.test.espresso.assertion.ViewAssertions.matches;
import static androidx.test.espresso.matcher.ViewMatchers.withClassName;
import static androidx.test.espresso.matcher.ViewMatchers.withId;
import static androidx.test.espresso.matcher.ViewMatchers.withText;

/**
 * @author  HDunn, Modifed on 4/30/21.
 */

public class DateTimePickerTest extends TestBase {

    @Test
    public void testSetDate() {
        int year = 2020;
        int month = 11;
        int day = 15;

        // From Main navigate to date time pickers
        onView(withId(R.id.pickers_button)).perform(click());

        onView(withId(R.id.date_picker_button)).perform(click());
        onView(withClassName(Matchers.equalTo(DatePicker.class.getName()))).perform(PickerActions.setDate(year, month + 1, day));
        onView(withId(android.R.id.button1)).perform(click());

        onView(withId(R.id.status)).check(matches(withText(year + "/" + month + "/" + day)));
    }

    @Test
    public void testSetTime() {
        int hour = 10;
        int minutes = 59;

        // From Main navigate to date time pickers
        onView(withId(R.id.pickers_button)).perform(click());

        onView(withId(R.id.time_picker_button)).perform(click());
        onView(withClassName(Matchers.equalTo(TimePicker.class.getName()))).perform(PickerActions.setTime(hour, minutes));
        onView(withId(android.R.id.button1)).perform(click());

        onView(withId(R.id.status)).check(matches(withText(hour + ":" + minutes)));
    }
}

File: app/src/androidTest/java/com/vgrec/espressoexamples/DateTimePickerTest.kt
package com.vgrec.espressoexamples

import android.widget.DatePicker
import android.widget.TimePicker
import androidx.test.espresso.Espresso
import androidx.test.espresso.action.ViewActions
import androidx.test.espresso.assertion.ViewAssertions
import androidx.test.espresso.contrib.PickerActions
import androidx.test.espresso.matcher.ViewMatchers
import com.vgrec.espressoexamples.bases.TestBase
import org.hamcrest.Matchers
import org.junit.Test

/**
 * @author  HDunn, Modified on 4/30/21.
 */
class DateTimePickerTest : TestBase() {
    @Test
    fun testSetDate() {
        val year = 2020
        val month = 11
        val day = 15

        // From Main navigate to date time pickers
        Espresso.onView(ViewMatchers.withId(R.id.pickers_button)).perform(ViewActions.click())
        Espresso.onView(ViewMatchers.withId(R.id.date_picker_button)).perform(ViewActions.click())
        Espresso.onView(ViewMatchers.withClassName(Matchers.equalTo(DatePicker::class.java.name))).perform(PickerActions.setDate(year, month + 1, day))
        Espresso.onView(ViewMatchers.withId(android.R.id.button1)).perform(ViewActions.click())
        Espresso.onView(ViewMatchers.withId(R.id.status)).check(ViewAssertions.matches(ViewMatchers.withText("$year/$month/$day")))
    }

    @Test
    fun testSetTime() {
        val hour = 10
        val minutes = 59

        // From Main navigate to date time pickers
        Espresso.onView(ViewMatchers.withId(R.id.pickers_button)).perform(ViewActions.click())
        Espresso.onView(ViewMatchers.withId(R.id.time_picker_button)).perform(ViewActions.click())
        Espresso.onView(ViewMatchers.withClassName(Matchers.equalTo(TimePicker::class.java.name))).perform(PickerActions.setTime(hour, minutes))
        Espresso.onView(ViewMatchers.withId(android.R.id.button1)).perform(ViewActions.click())
        Espresso.onView(ViewMatchers.withId(R.id.status)).check(ViewAssertions.matches(ViewMatchers.withText("$hour:$minutes")))
    }
}
